src/adapters/DebugProtocolAdapter.ts

```diff
@@ -422,96 +468,91 @@ export class DebugProtocolAdapter {
+            for (let variable of response.variables) {
+                let value;
+                let variableType = variable.variableType;
+                if (variable.value === null) {
+                    value = 'roInvalid';
+                } else if (variableType === 'String') {
+                    value = `\"${variable.value}\"`;
+                } else {
+                    value = variable.value;
+                }
+
+                if (variableType === 'Subtyped_Object') {
+                    let parts = variable.value.split('; ');
+                    variableType = `${parts[0]} (${parts[1]})`;
+                } else if (variableType === 'AA') {
+                    variableType = 'AssociativeArray';
+                }
 
-                    let container = <EvaluateContainer>{
-                        name: expression,
-                        evaluateName: expression,
-                        variablePath: variablePath,
-                        type: variableType,
-                        value: value,
-                        keyType: variable.keyType,
-                        elementCount: variable.elementCount
-                    };
-
-                    if (!firstHandled && variablePath.length > 0) {
+                let container = <EvaluateContainer>{
+                    name: expression,
+                    evaluateName: expression,
+                    variablePath: variablePath,
+                    type: variableType,
+                    value: value,
+                    keyType: variable.keyType,
+                    elementCount: variable.elementCount
+                };
+
+                if (!firstHandled && variablePath.length > 0) {
+                    firstHandled = true;
+                    mainContainer = container;
+                } else {
+                    if (!firstHandled && variablePath.length === 0) {
+                        // If this is a scope request there will be no entries in the variable path
+                        // We will need to create a fake mainContainer
                         firstHandled = true;
-                        mainContainer = container;
+                        mainContainer = <EvaluateContainer>{
+                            name: expression,
+                            evaluateName: expression,
+                            variablePath: variablePath,
+                            type: '',
+                            value: null,
+                            keyType: 'String',
+                            elementCount: response.numVariables
+                        };
+                    }
+
+                    let pathAddition = mainContainer.keyType === 'Integer' ? children.length : variable.name;
+                    container.name = pathAddition.toString();
+                    if (mainContainer.evaluateName) {
+                        container.evaluateName = `${mainContainer.evaluateName}.["${pathAddition}"]`;
```

Comment:
Remove the dot in this expression. I'll die on this hill! 
```suggestion
                        container.evaluateName = `${mainContainer.evaluateName}["${pathAddition}"]`;
```